### ISSUE TITLE
Feature: LHN, avatar and header updates to reports, workspace chats and chat rooms tied to workspaces.

### ISSUE THREAD

**Issue by JmillsExpensify** [Expensify/App#20486]
While we recently implemented threads, tasks, and manual requests, we realized that threads followed one set of design patterns, while manual requests and tasks followed a different set of patterns. In an effort to clean this up for manual requests and IOUs – threads are already being implemented and tasks are being handled separately – this doc outlines a new, unified approach to ensure business use cases match the rest of the app. 

**LHN & Avatars**
<img width="487" alt="Screenshot 2023-06-08 at 21 18 23" src="https://github.com/Expensify/App/assets/1127863/a8ba0fc1-8f79-46c8-9d0d-de604c447cd4">

**Notes for completeness**: 
 - The **workspace chat** now uses a consistent avatar for members and admins, with the overarching goal being consistency and less confusion. Workspace chats use a 40x40 dual-avatar, with a larger workspace avatar and a smaller member avatar overlaid in the bottom-right. That said, the name of the workspace chat will continue to differ for admins, taking the form %memberName%, so that they’re able to find workspace chats for specific employees.
 - **Expense reports** use a 24x24 dual-avatar, with the rounded member avatar in the top-left and a rounded square workspace avatar in the bottom right. This dual avatar emphasizes the collaboration and shared nature of sending a request and eventually getting paid back. The title takes the form “%workspaceName% owes %amount%” (and updates to %workspaceName paid %amount% when settled). 
 - **IOU reports** will similarly use a 24x24 dual-avatar, though both members have a rounded avatar in contrast. The title is the same as expense reports.
 - **Requests** made to the workspace will feature a larger 32x32 rounded member avatar, as well as a smaller 20x20 rounded square workspace avatar in the case of expense requests. The workspace overlay helps distinguish requests to a workspace vs a DM, which has no extra overlay. The title takes the form “%amount% request for %merchant% | %comment%”. 
 - **Threads in workspace rooms/reports** will use a distinct “overlay” avatar from requests. A larger 40x40 circular member avatar as well as a smaller 20x20 rounded square workspace avatar overlaid in the bottom right. This similarly helps distinguish workspace threads from DMs, and emphasizes the member who made the parent comment. The title of the thread is the parent comment. 
 - **Threads in DMs/GroupDMs** will simply use the avatar of the parent message sender. 
 - **All chat rooms tied to workspaces** will feature the workspace avatar. This deprecates the chair and similar icons for user-created rooms. 

**Headers**
![image](https://github.com/Expensify/App/assets/1127863/4f925be4-6707-4126-a473-f92b5fc991b5)

**Notes for completeness**:
 - For the first line, the avatars and header follow the same convention as the LHN. 
 - For the second line, we use “From %parentChat% in %policyName%”.
 - The parent report is always hyperlinked, displayed in blue and can be clicked to return to the parent report. 

**iou/expenseReport subheaders*

- Remove the `To:` and `amount`
- reposition the `Pay` button
- modify the pay button copy, so it includes the report amount. i.e `Pay $50 with Expensify`

**Mobile (Old | New):**
![image](https://github.com/Expensify/App/assets/16232057/0360eeb6-8ad1-4673-9bf1-65c7f5fe2c19)

**Desktop (Old | New):**
![image](https://github.com/Expensify/App/assets/16232057/359bf186-8f36-437c-a3b6-7f871c6a49fe)

### Rollout Plan

- [x] Update the subheader and parent navigation on all report types ([PR](https://github.com/Expensify/App/pull/21092)) - @chiragsalian 
- [x] Update the avatars in all places they feature (LHN, Chat Headers, Report Previews) ([GH](https://github.com/Expensify/App/issues/19264)) - @grgia 
- [x] Update the iou/expenseReport header logic. - @grgia - ([GH](https://github.com/Expensify/App/issues/22333))
    - Remove the `To:` and `amount`
     - reposition the `Pay` button
     - modify the pay button copy, so it includes the report amount. i.e `Pay $50 with Expensify`
     - Add the "wide" reportTotal to the expanded scrollable header
- [x] Update the iou/expense request header logic. - @grgia ([GH](https://github.com/Expensify/App/issues/22333))

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01d3518465643567ea</li>
        <li>Upwork Job ID: 1666939604347977728</li>
        <li>Last Price Increase: 2023-06-08</li>
    </ul>
</details>

**Comment by JmillsExpensify** [comment]
Also @shawnborton did we decide one way or the other on all blue for the second line of the header rows? I still like the differentiated `From` but I think you had some doubts. Chime in now, because we're going to start implementing this soon!

**Comment by shawnborton** [comment]
I'm cool with leaving From in gray to differentiate it, I agree it does look a bit nicer.

**Comment by grgia** [comment]
@shawnborton are we good to combine this issue with https://github.com/Expensify/App/issues/19264 ?

**Comment by trjExpensify** [comment]
>I'm cool with leaving From in gray to differentiate it, I agree it does look a bit nicer.

I agree. @chiragsalian, want to add that as a commit to [your PR](https://github.com/Expensify/App/pull/20484) before it gets reviewed and merged?

**Comment by trjExpensify** [comment]
>@shawnborton are we good to combine this issue with https://github.com/Expensify/App/issues/19264 ?

I don't see why not. We can probably use this issue as a bit of an overarching tracker, and we can add those to the OP. 

Breaking that down, is this what we need to complete the refresh across all report types? 

- [ ] Update the subheader and parent navigation on all report types ([GH](https://github.com/Expensify/Expensify/issues/290078)) - @chiragsalian 
- [ ] Update the avatars in all places they feature (LHN, Chat Headers, Report Previews) ([GH](https://github.com/Expensify/App/issues/19264)) - @grgia 
- [ ] Deprecate the iou/expenseReport header logic. 
    - Remove the `To:` and `amount`
     - reposition the `Pay` button
     - modify the pay button copy, so it includes the report amount. i.e `Pay $50 with Expensify`

On that third one, that's these updates in the header side-by-side:  

**Mobile (Old | New):**
![image](https://github.com/Expensify/App/assets/16232057/0360eeb6-8ad1-4673-9bf1-65c7f5fe2c19)

**Desktop (Old | New):**
![image](https://github.com/Expensify/App/assets/16232057/359bf186-8f36-437c-a3b6-7f871c6a49fe)


Note: I've purposefully left out the new sub-total header stuff, I think that only needs to come first when we bring in card transactions and hold.

 
Does that sound about right to everyone? Missed anything, anything to add?

**Comment by grgia** [comment]
Avatar portion should be done monday, just debugging a failing Jest test

**Comment by grgia** [comment]
On that, Code is done otherwise! PR draft here-  https://github.com/Expensify/App/pull/20512

**Comment by shawnborton** [comment]
Amazing! And thank you Tom for breaking it down like that.

**Comment by grgia** [comment]
Could I get clarification on QA steps for creating an expense report vs expense request?
I'm able to make an expense request in LHN, but I'm not sure how we differentiate a normal workspace chat vs an expense report chat

**Comment by trjExpensify** [comment]
pre-requisite: Have the account(s) on the `policyExpenseChat` beta to get access to a workspace chat.  

>I'm able to make an expense request in LHN, but I'm not sure how we differentiate a normal workspace chat vs an expense report chat

In the LHN:

1. The workspace chat report will be called `Shawn Borton`
2. The expenseReport will be called `Duraflame owes $100.00` 
3. The request will be called `$100.00 request for Thursday Lunch`

So as suggested QA steps, maybe like so... 

1. Make sure you have a couple of accounts on the `policyExpenseChat` beta (or use expensifail.com)
2. Create a free plan workspace > set a custom workspace avatar
3. Add a member to the workspace 
4. Now log-in as the member, set a custom avatar 
5. Head to your workspace chat in the LHN 
6. **Verify** the avatar is the custom one you set with the custom workspace avatar overlaid in the bottom right
7. Navigate to the workspace chat > click the `+` icon > `request money` > create and request $100.00 from the workspace > land in the workspace chat
8. **Verify** the report preview uses a 24x24 dual-avatar, with the rounded member avatar in the top-;eft and a rounded square workspace avatar in the bottom right
9. Click the report preview that was added to the workspace chat 
10. **Verify** in the LHN the expenseReport avatar use a 24x24 dual-avatar, with the rounded member avatar in the top-left and a rounded square workspace avatar in the bottom right.
11. On the expenseReport, click into the request to access the request view
12. **Verify** the request in the LHN uses a larger 32x32 rounded member avatar, as well as a smaller 20x20 rounded square workspace avatar overlaid in the bottom right.

**Comment by trjExpensify** [comment]
>@trjExpensify can you please guide me which page is expense request, which page is expense report in video? Since I will be reviewing PR.

The expenseReport is the one with the pay button in the header seen at 0:03. 
The request is the one with the details (amount, description, date) seen at 0.15.

**Comment by grgia** [comment]
I have the avatar PR up, will start working on all of the headers next

**Comment by grgia** [comment]
Notes for titles:  

- if **workspace chat** 
      isAdmin?
            `%memberName%`
      else 
            `%workspaceName%`
- if **Expense reports** 
      isSettled?
            `%workspaceName paid %amount%`
      else
             `%workspaceName% owes %amount%`
- if **IOU reports** 
      isSettled?
            `%payer% paid %amount%`
      else
            `%payer% owes %amount%`
- if **Requests** made to the workspace 
      `%amount% request for %merchant% | %comment%`
- if **Threads in workspace rooms/reports** 
      `%parent comment%`
- if **Threads in DMs/GroupDMs**
      `%parent comment%`
- if **Chat room**
      `%roomName%`

**Comment by grgia** [comment]
@trjExpensify / @JmillsExpensify do we have a mockup that features the **Request** with header pattern `%amount% request for %merchant% | %comment%` ? I want to clarify what the `comment` is in this case

**Comment by grgia** [comment]
After looking at the current logic, I believe that the only title that will need to be updated in the code is the above **Request**

**Comment by grgia** [comment]
Actually, I think the title header logic is currently correct. I understand that the `comment` is the request description. What would `merchant` come from? 

@trjExpensify 

Also, pertaining to https://github.com/Expensify/App/issues/20486#issuecomment-1584397254 is there an issue for deprecating the IOU logic? Since there's no work to be done for headers, I have some bandwidth to work on that if it's not already being worked on

**Comment by trjExpensify** [comment]
👋 yo yo! I'm losing sight of how and where these changes are being implemented exactly. Can we get the rollout plan in the OP updated so we're all aligned and following it? 

>Update the subheader and parent navigation on all report types (https://github.com/Expensify/Expensify/issues/290078) - @chiragsalian

- @chiragsalian your linked PR in the OP has been deployed, but the subheaders don't seem to have been updated for anything other than "message" threads on staging. Can you link your PR for the other report types, where is that one? 

>Update the avatars in all places they feature (LHN, Chat Headers, Report Previews) (https://github.com/Expensify/App/issues/19264) - @grgia

- @grgia you have a linked PR for avatars [here](https://github.com/Expensify/App/pull/20512), but looking at [the review checklist](https://github.com/Expensify/App/pull/20512#issuecomment-1613523926), the avatars in the headers aren't updated (and I can't see the report preview in the workspace chat to confirm), only the avatars in the LHN. Does that mean there's a follow-up for it somewhere, I'm curious why the changes wouldn't apply to all places the avatars feature as one? 

>Deprecate the iou/expenseReport header logic.
> - Remove the To: and amount
> - reposition the Pay button
> - modify the pay button copy, so it includes the report amount. i.e Pay $50 with Expensify

- There isn't a PR linked for this one currently. _I think_ @grgia you said you were [taking it here](https://github.com/Expensify/App/issues/20486#issuecomment-1594607519) is that right, can you link it? 
- One thing we'll probably want to add to this is the "wide" total in the new header, as that has since been agreed to always be shown now (not just when there's subtotals). 

Somewhat related, but [this PR](https://github.com/Expensify/App/pull/21918) has popped up for making the existing headers scrollable. Is that going to cause conflicts/will we rip it out with some of the items above? If so, we should probably halt it.

**Comment by grgia** [comment]
@trjExpensify I think we need to split the remaining parts into separate issues.

> @grgia you have a linked PR for avatars https://github.com/Expensify/App/pull/20512, but looking at https://github.com/Expensify/App/pull/20512#issuecomment-1613523926, the avatars in the headers aren't updated (and I can't see the report preview in the workspace chat to confirm), only the avatars in the LHN. Does that mean there's a follow-up for it somewhere, I'm curious why the changes wouldn't apply to all places the avatars feature as one?

I will do a second PR for updating the headers. Can I get the mockups for the final header designs for mobile/desktop? 

> Deprecate the iou/expenseReport header logic.
> Remove the To: and amount
> reposition the Pay button
> modify the pay button copy, so it includes the report amount. i.e Pay $50 with Expensify
> There isn't a PR linked for this one currently. I think @grgia you said you were https://github.com/Expensify/App/issues/20486#issuecomment-1594607519 is that right, can you link it?
> One thing we'll probably want to add to this is the "wide" total in the new header, as that has since been agreed to always be shown now (not just when there's subtotals).

This will be a part of the above PR, can I again get final mockups for mobile/desktop


> Somewhat related, but https://github.com/Expensify/App/pull/21918 has popped up for making the existing headers scrollable. Is that going to cause conflicts/will we rip it out with some of the items above? If so, we should probably halt it.

I don't think that PR will cause any problems

**Comment by trjExpensify** [comment]
>I will do a second PR for updating the headers. Can I get the mockups for the final header designs for mobile/desktop?
>This will be a part of the above PR, can I again get final mockups for mobile/desktop

Gotcha, so they are separate. The notable change is the "Total" piece and background colour: 

![image](https://github.com/Expensify/App/assets/16232057/81bf5874-29c0-4df2-92ec-38aa9d25e027)

>And money request header won't be "header" anymore after making it scrollable along with report actions.

Okay cool, but in that PR it still looks like it isn't using the correct background, so it does appear like a header still at this point. Sounds like Georgia's follow-up PR will change that on the iou/expenseReport view. 

One thing it doesn't seem like is accounted for yet is the change to the request view, which similarly won't have the request details fixed to the top in a different background colour. That one should look like the below, so we should probably add a new bullet in the OP for this: 

![image](https://github.com/Expensify/App/assets/16232057/88debf1b-68fb-4820-9e5f-07c127a31abc)

**Comment by chiragsalian** [comment]
> Update the subheader and parent navigation on all report types (https://github.com/Expensify/Expensify/issues/290078) - @chiragsalian

Thats the old GH issue link. It was only tackled for regular chat reports. Since then jason/db asked me to update it for all chat reports and this is the GH issue - https://github.com/Expensify/App/issues/20844.

The PR for that issue went to staging 28minutes back 🙂 so feel free to test it out.

**Comment by trjExpensify** [comment]
>The PR for that issue went to staging 28minutes back 🙂

Dope! I've linked the PR in the OP and crossed that off 👍

**Comment by grgia** [comment]
I just want to make sure I understand this comment right before I work on these today- https://github.com/Expensify/App/issues/20486#issuecomment-1622110749

1. These UI changes apply to both IOU reports/requests and Expense reports/requests right?
2. The main changes are:
- [ ] The header is being updated to follow the normal header pattern, the pay button is moved inside header on web and under header on mobile
- [ ] The back button is added to the header
- [ ] These reports use the background image
- [ ] On Expense/IOU REPORTS, we add a divider line and the total. (question: do only open requests go here? what happens if a user has a history of 10 requests, do they need to scroll all the way up to see the total?)
- [ ] On Expense/IOU REQUESTS, we add a divider line and display the information above the divider line


For reference I'm adding the current page views (note avatars are to be updated):
IOU Report
<img width="300" alt="image" src="https://github.com/Expensify/App/assets/38015950/b10559f8-6c26-44c7-896c-e832830c304c"> 
IOU Request
<img width="300" alt="image" src="https://github.com/Expensify/App/assets/38015950/58bfc1da-0238-4d73-b925-c949c357c7d8">


Expense Report
<img width="300" alt="image" src="https://github.com/Expensify/App/assets/38015950/8b051789-e9fc-49b0-ae6a-34175f48728c">
Expense Request
<img width="300" alt="image" src="https://github.com/Expensify/App/assets/38015950/1c808fd4-2472-49b7-926b-8349d15d6bc1">

**Comment by grgia** [comment]
I created the above two issues so we'll be able to track progress on these two views since this issue is covering a handle of changes

**Comment by trjExpensify** [comment]
>These UI changes apply to both IOU reports/requests and Expense reports/requests right?

Yup! 

>The main changes are:
> - [ ] The header is being updated to follow the normal header pattern, the pay button is moved inside header on web and under header on mobile
> - [ ] The back button is added to the header
> - [ ] These reports use the background image
> - [ ] On Expense/IOU REPORTS, we add a divider line and the total. (question: do only open requests go here? what happens if a user has a history of 10 requests, do they need to scroll all the way up to see the total?)
> - [ ] On Expense/IOU REQUESTS, we add a divider line and display the information above the divider line

adding to this.. 

> - [ ] On Expense/IOU REPORTS the pay button now moved to the header includes the reportTotal i.e `Pay $10.00 with Expensify` / `Pay $10.00 elsewhere` / `Pay $10.00 using Paypal.me`. That way, the total to pay is always in view. 

>The back button is added to the header

On mobile yep, but it's already there in the header, so I don't think this is an addition is it?

>On Expense/IOU REPORTS, we add a divider line and the total. (question: do only open requests go here? what happens if a user has a history of 10 requests, do they need to scroll all the way up to see the total?)

Not sure I understand the first part of the question sorry, can you rephrase? As for the second part of the question, the reportTotal is always in view within the pay button in the header.

**Comment by grgia** [comment]
> Not sure I understand the first part of the question sorry, can you rephrase? As for the second part of the question, the reportTotal is always in view within the pay button in the header.

![image](https://github.com/Expensify/App/assets/38015950/cc7ecf87-bfb4-4497-9dbb-9a8300659cfb)

So for an  _Expense Report View_ you could have multiple Expense Requests underneath right? In this example, we have just one, the $136 charge. Say we have 5 more Expense Requests underneath the $136 charge. Normally in chat, we open to the bottom of the chat, so wouldn't the divider and total be pushed off-screen, similar to how the background image is pushed off-screen as you send more chats?

**Comment by grgia** [comment]
One more question, @trjExpensify with the updated headers- do we show any UI for if the report/expense is settled? Currently, we show a green checkmark next to the amount in the header. 

<img width="115" alt="image" src="https://github.com/Expensify/App/assets/38015950/9e1e5275-8d02-47dd-8cce-37b40995611a">

**Comment by trjExpensify** [comment]
>So for an Expense Report View you could have multiple Expense Requests underneath right? In this example, we have just one, the $136 charge. Say we have 5 more Expense Requests underneath the $136 charge. Normally in chat, we open to the bottom of the chat, so wouldn't the divider and total be pushed off-screen, similar to how the background image is pushed off-screen as you send more chats?

That's correct, the "wide" total would be pushed off the page as more report actions get added to the report. The amount to pay would still however be visible in the pay button in the header atop the report, like this for example: 

![image](https://github.com/Expensify/App/assets/16232057/684ea9fb-594c-4527-b633-75c6b49c1301)


>Currently, we show a green checkmark next to the amount in the header.

Great question! I don't think we planned to remove the checkmark when paid from this "wide" total that replaces the current amount in the fixed header. @shawnborton can you confirm, or did we want to remove that?

**Comment by shawnborton** [comment]
I think it makes sense to keep it, though I wonder if we should put it on the left side of the total so we keep the nice right-edge alignment?

**Comment by trjExpensify** [comment]
Nice, that makes sense to me. 👍

**Comment by grgia** [comment]
@trjExpensify @shawnborton great, that makes sense with the total in the button! Could I get a mockup for the checkmark place on web/mobile? I want to make sure I'm on the same page there

**Comment by grgia** [comment]
@shawnborton 
![image](https://github.com/Expensify/App/assets/38015950/f00b2a3c-5711-465c-b710-1bd8588827c9)
Is it intentional to remove the divider line under the header only on mobile?

**Comment by shawnborton** [comment]
Definitely not, that was probably an accident. Let's keep the divider line there.

**Comment by trjExpensify** [comment]
This whatcha' thinking for the total yeah? 

<img width="354" alt="image" src="https://github.com/Expensify/App/assets/16232057/19808826-edcc-4697-8753-d544ebee52ad">

**Comment by shawnborton** [comment]
Yeah I think that works!

**Comment by grgia** [comment]
**OVERALL UPDATE**:

I made a few PRs to get started on the headers this weekend- one for matching the avatars (https://github.com/Expensify/App/pull/22467)  and one for the expense _REPORT_ view (https://github.com/Expensify/App/pull/22484) 

Both are nearly done and just need screenshots/QA steps and I think I should be able to get most of the Expense _REQUEST_ view done today as well.

---

**DROP DOWN BUTTON QUESTION**:

---

@shawnborton - We modify the dropdown button on this page, can you confirm whether this design will be used everywhere or just on this page? (changed to medium sized button (smaller height), remove dark divider line)

before:
<img width="300" alt="image" src="https://github.com/Expensify/App/assets/38015950/3c256d23-47ee-420f-831a-86fcb968e48e">

after (in mockup):
<img width="300" alt="image" src="https://github.com/Expensify/App/assets/38015950/0f9e648e-7288-4118-8d39-ba19fded0cff">

I have a few follow-up questions about hover and press actions, so my plan is to create an issue and tackle this button separately. Here's the issue- https://github.com/Expensify/App/issues/22540

---

**THREE DOTS QUESTION**:

---

@trjExpensify  - What happens when you click the three dots? We currently omit the three dots on report pages and only show on the individual transactions. Is it intentional to add it here? (The default option is delete)

<img src="https://github.com/Expensify/App/assets/38015950/d47324ac-7a53-4991-8517-8aa1abf3e0f5" width="500">

**Comment by trjExpensify** [comment]
>@trjExpensify - What happens when you click the three dots? We currently omit the three dots on report pages and only show on the individual transactions. Is it intentional to add it here? (The default option is delete)

Oh hm, you're right. @shawnborton are we moving the pin action into that overflow menu here, yeah? I believe that was the intention, in which case we should add it as a menu item within both the expense/iouReport view and request view.

**Comment by shawnborton** [comment]
Yeah, the intention was to move the pin action into the overflow menu. I like keeping things consistent and training the user that on basically every type of view (chat, task, expense, report, etc) you have an overflow menu that has additional options. So even if the pin is the only option available in some cases, I do like keeping things consistent by using the overflow menu.

**Comment by grgia** [comment]
I have both PRs MVP done for these two views. @shawnborton @trjExpensify could we sync up and make sure I'm not missing any steps in these PRs? 

> the intention was to move the pin action into the overflow menu

Is there a separate issue or design doc section for the pin changes?

Report
https://github.com/Expensify/App/pull/22484

Request
https://github.com/Expensify/App/pull/22562

**Comment by shawnborton** [comment]
I think we missed putting this into an issue or design doc. That being said, happy to sync up and make sure we're all on the same page!

**Comment by shawnborton** [comment]
But yeah, the idea is that we basically give every top chat header (the thing with the avatar + user/room/task/expense/report name) an overflow menu. The pin icon will live behind the overflow menu, and sometimes we'll also have a call icon, a delete icon, etc.

**Comment by grgia** [comment]
EOD Update:

Major PRs Ready For Review:
🟢 Match LHN Avatars to Header Avatars https://github.com/Expensify/App/pull/22467
🟢 Expense Report Headers https://github.com/Expensify/App/pull/22484
🟢 Expense Request Headers https://github.com/Expensify/App/pull/22562

Follow-Up Issues I will be tackling separately as they affect more views: (Note, I may make some of these external)
🟡 Move Pin Logic into Context Menu - https://github.com/Expensify/App/issues/22635
🟡 Update DropDown button style - https://github.com/Expensify/App/issues/22540
🟡 Update Header Link Style - https://github.com/Expensify/App/issues/22641

**Comment by trjExpensify** [comment]
I think this should be back on `weekly` until we can close it out.

**Comment by JmillsExpensify** [comment]
Yeah, agreed. We're down to clean-up now.

**Comment by JmillsExpensify** [comment]
Given that we have an issue for everything that remains from here, and related because all those are already included in the wave3 project, I'm inclined to close this larger issue out. In fact, I'm going to go ahead and do that, though please re-open if I'm missing something.
